cpp/src/parquet/encryption/internal_file_decryptor.cc

```diff
@@ -147,13 +141,10 @@ std::shared_ptr<Decryptor> InternalFileDecryptor::GetFooterDecryptor(
   std::shared_ptr<encryption::AesDecryptor> aes_metadata_decryptor;
   std::shared_ptr<encryption::AesDecryptor> aes_data_decryptor;
 
-  {
-    std::lock_guard<std::mutex> lock(mutex_);
-    aes_metadata_decryptor = encryption::AesDecryptor::Make(
-        algorithm_, key_len, /*metadata=*/true, &all_decryptors_);
-    aes_data_decryptor = encryption::AesDecryptor::Make(
-        algorithm_, key_len, /*metadata=*/false, &all_decryptors_);
-  }
+  aes_metadata_decryptor =
+      encryption::AesDecryptor::Make(algorithm_, key_len, /*metadata=*/true);
+  aes_data_decryptor =
+      encryption::AesDecryptor::Make(algorithm_, key_len, /*metadata=*/false);
```

Comment:
I still don't think this function is correct, because it still caches the decryptors. It's the footer key that should be cached in `InternalFileDecryptor`, not the decryptors themselves.

(also, please let's reuse `GetFooterKey` instead of duplicating its code)